pyproject.toml

```diff
@@ -40,6 +36,22 @@ classifiers = [
     'Operating System :: Unix',
     'Operating System :: MacOS',
 ]
+# License info:
+#   - The main NumPy project license is BSD-3-Clause.
+#   - The SPDX expression for license and the included license files can be
+#     found near the top of ./meson.build
+#   - The SPDX license expression reflects installed numpy packages when built
+#     from source (e.g., with `python -m build --wheel`), with no vendoring.
+#   - That SPDX expression is therefore incomplete for:
+#     (a) sdists - see the comment on license_files near the top of ./meson.build
+#     (b) wheels on PyPI - most wheels include vendored libraries with additional licenses:
+#         - libopenblas : BSD-3-Clause AND BSD-3-Clause-Attribution (all except arm64 macOS>=14)
+#         - libgfortran : GPL-3.0-with-GCC-exception                (all except arm64 macOS>=14)
+#         - libquadmath : LGPL-2.1-or-later                         (all except arm64 macOS>=14 and Windows)
+#         The licenses for these vendored components are dynamically included
+#         in the build process for PyPI wheels.
+# TODO: patch SPDX expression in PyPI wheel build process
+dynamic = ["license", "license-files"]
```

Comment:
Argh never mind this part, this is unnecessary (I was thinking about it for SciPy, there it is useful to handle meson subprojects correctly). I'll fix this tonight.

The rest of this PR can be reviewed for completeness and correctness of license inclusion.